### PR TITLE
python310Packages.accelerate: init at 0.16.0

### PR DESCRIPTION
**Pull request by YodaEmbedding**
###### Description of changes

Initialize `python3Packages.accelerate`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/accelerate/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  propagatedBuildInputs = [
+    numpy
+    packaging
+    psutil
+    pytorch
+    pyyaml
+  ];
+
+  nativeCheckInputs = [
+    pytest
```
Comment (dotlambda) [inline pkgs/development/python-modules/accelerate/default.nix:30]:
```suggestion
    pytestCheckHook
```

File: pkgs/development/python-modules/accelerate/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  ];
+
+  meta = with lib; {
+    description = "A simple way to train and use PyTorch models with multi-GPU, TPU, mixed-precision";
+    homepage = "https://github.com/huggingface/accelerate";
+    license = with lib.licenses; [ asl20 ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/accelerate/default.nix:36]:
```suggestion
    license = lib.licenses.asl20;
```

File: pkgs/development/python-modules/accelerate/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "accelerate";
+  version = "0.16.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-0T4w8+bev7Rsrae5Ma+FVgYZtqaoOdDK/uq27XxqSY0=";
```
Comment (dotlambda) [inline pkgs/development/python-modules/accelerate/default.nix:18]:
```suggestion
    hash = "sha256-0T4w8+bev7Rsrae5Ma+FVgYZtqaoOdDK/uq27XxqSY0=";
```

**Review by dotlambda — COMMENTED** [review]
Please add `format` and `pythonImportsCheck`.

File: pkgs/development/python-modules/accelerate/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, numpy
+, packaging
+, psutil
+, pytorch
```
Comment (dotlambda) [inline pkgs/development/python-modules/accelerate/default.nix:7]:
```suggestion
, torch
```
`pytorch` is just an alias

**Review by dotlambda — CHANGES_REQUESTED** [review]
Tests fail with
```
ModuleNotFoundError: No module named 'evaluate'`
```
and
```
ModuleNotFoundError: No module named 'datasets'
```
See also https://github.com/huggingface/accelerate/blob/main/setup.py#L22

**Comment by dotlambda** [comment]
If you call your commit `python310Packages.accelerate: init at 0.16.0` then ofborg will automatically build the package for you.

**Comment by kira-bruneau** [comment]
Added in https://github.com/NixOS/nixpkgs/commit/71e8e2148d920ba6a1db6f64e94cf135620c4923

**Pull request closed without merging**